src/ORM/UnionList.php

```diff
@@ -0,0 +1,257 @@
+     */
+    public function __construct(array $lists)
+    {
+        foreach ($lists as $list) {
+            $this->lists[] = clone $list;
+        }
+        parent::__construct();
+    }
+
+    /**
+     * Return the total number of items in each SS_List.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        $count = 0;
+        foreach ($this->lists as $list) {
+            $count += $list->count();
+        }
+        return $count;
+    }
+
+     /**
+     * Returns true if one of the SS_List's have an item.
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        foreach ($this->lists as $list) {
+            if (method_exists($list, 'exists') && $list->exists()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Iterate over each SS_List, one after the other.
+     *
+     * @return ArrayIterator
+     */
+    public function getIterator()
+    {
+        foreach ($this->lists as $list) {
+            foreach ($list as $record) {
+                yield $record;
+            }
+        }
+    }
+
+    /**
+     * Get array of each record in each SS_List.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $result = array();
+        foreach ($this->lists as $list) {
+            foreach ($list as $record) {
+                $result[] = $record;
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Walks the list using the specified callback
+     *
+     * @param callable $callback
+     * @return $this
+     */
+    public function each($callback)
+    {
+        foreach ($this as $item) {
+            $callback($item);
+        }
+        return $this;
+    }
+
+    public function debug()
+    {
+        $val = "<h2>" . $this->class . "</h2><ul>";
+        foreach ($this->toNestedArray() as $item) {
+            $val .= "<li style=\"list-style-type: disc; margin-left: 20px\">" . Debug::text($item) . "</li>";
+        }
+        $val .= "</ul>";
+        return $val;
+    }
+
+    /**
+     * Return this list as an array and every object it as an sub array as well
+     *
+     * @return array
+     */
+    public function toNestedArray()
+    {
+        $result = array();
+        foreach ($this->lists as $list) {
+            foreach ($list as $item) {
+                if (is_object($item)) {
+                    if (method_exists($item, 'toMap')) {
+                        $result[] = $item->toMap();
+                    } else {
+                        $result[] = (array) $item;
+                    }
+                } else {
+                    $result[] = $item;
+                }
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Get first item in the first SS_List.
+     *
+     * @return array
+     */
+    public function first()
+    {
+        $list = reset($this->lists);
+        return $list ? $list->first() : null;
```

Comment:
What if the first list didn't have any items but the second list did? Then this function would return null, but that wouldn't be correct. It should probably loop through all of the lists and return the first result in any list.